### PR TITLE
[eda][docs] Tutorials and UX improvements

### PR DESCRIPTION
**Pull request by gradientsky**
*Description of changes:*

See docs here: https://auto.gluon.ai/2023_01_25_eda_docs/tutorials/eda/index.html

- [doc]Missing values analysis + visualization/formatting fixes
- [doc] analyze_interaction showcase tutorial
- [doc] Added Covariate Shift Analysis
- [doc]Added quick-fit tutorial
- [doc] Target variable analysis tutorial
- [fix] Formatting fixes for shift
- [fix] Removed not useful information
- [CI] resolve tox-awscli dependencies conflict

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/tutorials/eda/eda-auto-analyze-interaction.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The average `Fare` closest to `$80` are in the `C` `Embarked` values where `Pclass` is `1`. Let's fill in the missing
+values as `C`.
+
+## Using Interaction Charts To Learn Information About the Data
+
+```{.python .input}
+auto.analyze_interaction(x='Pclass', y='Survived', train_data=df_train, test_data=df_test)
+```
+
+It looks like `63%` first class passengers survived; `48%` second class and only `24%` third class passenger survived.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-analyze-interaction.md:53]: Here's few nits: "It looks like `63%` of first class passengers survived, while `48%` of second class and only `24%` of third class passengers survived."

File: docs/tutorials/eda/eda-auto-analyze-interaction.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+```{.python .input}
+auto.analyze_interaction(x='Fare', hue='Survived', train_data=df_train, test_data=df_test, chart_args=dict(fill=True))
+```
+
+```{.python .input}
+auto.analyze_interaction(x='Age', hue='Survived', train_data=df_train, test_data=df_test)
+```
+
+The very left part of the distribution on this chart possibility hints that children and infants were the priority.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-analyze-interaction.md:64]: "possibly"

File: docs/tutorials/eda/eda-auto-covariate-shift.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+```{.python .input}
+import autogluon.eda.auto as auto
+
+auto.covariate_shift_detection(train_data=df_train, test_data=df_test, label=target_col)
+```
+
+The detector found that `Name` and `PassengerId` with a very high certainty (`roc_auc` is `1`) can distinguish if the
+row came from a train or test parts of the dataset. We'll ignore `Name` for now - it's importance is relatively low and
+first look at `PassengerId`. The graph shows that the feature is uniformly distributed across different ranges between
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-covariate-shift.md:35]: I had to re-read this sentence a few time:
> We'll ignore `Name` for now - it's importance is relatively low and first look at `PassengerId`.

Maybe something like this instead:
> We'll ignore `Name` for now - it's importance is relatively low, and instead we'll look first at `PassengerId`.

File: docs/tutorials/eda/eda-auto-dataset-overview.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+To showcase near duplicates detection functionality, let's add a duplicated column: 
+```{.python .input}
+df_train['Fare_duplicate'] = df_train['Fare']
+df_test['Fare_duplicate'] = df_test['Fare']
+```
+
+The report consists of multiple parts: statistical information overview enriched with feature types detection and
+missing values counts.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-dataset-overview.md:32]: "value"

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+# EDA: Automated Quick Model Fit
+
+The purpose of this feature is to provide a quick and easy way to obtain a preliminary understanding of the
+relationships between the target variable and the independent variables in a dataset.
+
+This functionality automatically splits the training data, performs a simple regression or classification model to the
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:6]: "fits a simple"

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+data and generating insights: model performance metrics, feature importance and prediction result insights.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:7]: "generates"

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+To inspect the prediction quality, a confusion matrix displayed for classification problems and scatter plot for
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:9]: "is displayed"

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+regression problems. Both representation allow to see the difference between actual and predicted values.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:10]: "allow the user to see"

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+The insights highlight two subsets of the model predictions:
+
+- predictions with the largest classification error. Rows listed in this section are candidates for inspection why the
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:14]: "inspecting"

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  model made the mistakes
+- predictions with the least distance from the other class. Rows in this category are most 'undecided'. They are useful
+  as an examples of data which is close to a decision boundary between the classes. The model would benefit from having
+  more data for a similar cases.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:18]: "for similar cases."

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+## Classification Example
+
+We will start with getting titanic dataset and performing a quick one-line overview to get the information.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:22]: "the titanic dataset"

File: docs/tutorials/eda/eda-auto-quick-fit.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+auto.quick_fit(df_train, target_col, show_feature_importance_barplots=True)
+```
+
+## Regression Example
+
+In the previous section we tried a classification example. Let's try a regression. It has a few differences.
+We are also going to store the fitted model by specifying `return_state` and `save_model_to_state` parameters.
+This will allow us to use the model to predict test values later.
+
+It is a large dataset, so we'll just keep only few columns for this tutorial.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-quick-fit.md:41]: "we'll just keep a few columns" or "we'll keep only a few columns"

File: docs/tutorials/eda/eda-auto-target-analysis.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
+# EDA: Automated Target Variable Analysis
+
+In this section we explore automated dataset overview functionality.
+
+Automated target variable analysis aims to automatically analyze and summarize the variable we are trying to predict (
+label). The goal of this analysis is to take a deeper look into target variable structure and its relationship with
+other important variables in the dataset.
+
+To simplify outliers and useful patterns discovery. This functionality introducing components which allow generating
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-target-analysis.md:9]: "To simplify outliers and useful patterns discovery." is a sentence fragment. I don't know what you want to do with it, but as a reader it's not clear how to read this.

Also, "introducing" -> "introduces"

File: docs/tutorials/eda/eda-auto-target-analysis.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+descriptive statistics, visualizing the target distribution and relationships between the target variable and other
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-target-analysis.md:10]: Either you should replace the "," with an "and" since there are technically only two items you are listing ("generating ... and visualizing ...") or put a verb in front of "relationships" e.g. (generating ..., visualizing ... and describing relationships...)

File: docs/tutorials/eda/eda-auto-target-analysis.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+variables in the dataset.
+
+## Classification Example
+
+We will start with getting titanic dataset and performing a quick one-line overview to get the information.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-target-analysis.md:15]: "the titanic dataset"

File: docs/tutorials/eda/eda-auto-target-analysis.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+The next component is feature distribution visualization. This is helpful for choosing data transformations and/or
+model selection. For regression tasks, the framework automatically fits multiple distributions available in scipy. The
+distributions with the best fit will be displayed on the chart. Distributions information will be displayed below the
+chart.
+
+Next, the report will provide correlation analysis focusing only on highly-correlated features and visualization of
+their relationships with the target.
+
+The final part of the analysis is label insights
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-target-analysis.md:39]: Should this be a title or incorporated into the next paragraph or at least have a '.' if it is meant to be a stand-alone sentence?

File: docs/tutorials/eda/eda-auto-target-analysis.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+The last chart is a feature distance. It measures the similarity between features in a dataset. For example, if two
+variables are almost identical, their feature distance will be small. Understanding feature distance is useful in
+feature
+selection, where it can be used to identify which variables are redundant and should be considered for removal.
```
Comment (gidler) [inline docs/tutorials/eda/eda-auto-target-analysis.md:44]: nit: the line break is weird here

**Review by gidler — COMMENTED** [review]
The technical content here is excellent. I had a handful of word-smithing callouts.

File: eda/src/autogluon/eda/auto/simple.py [hunk 18]
```diff
@@ -820,3 +826,85 @@ def _render_distribution_fit_information_if_available(state, label) -> Optional[
...
+        One of the following visualization types:
+        - matrix - nullity matrix is a data-dense display which lets you quickly visually pick out patterns in data completion
+            This visualization will comfortably accommodate up to 50 labelled variables.
+            Past that range labels begin to overlap or become unreadable, and by default large displays omit them.
+        - bar - visualizes how many rows are non-null vs null in the column. Logarithmic scale can by specifying `log=True` in `kwargs`
+        - heatmap - correlation heatmap measures nullity correlation: how strongly the presence or absence of one
+            variable affects the presence of another. Nullity correlation ranges from -1
+            (if one variable appears the other definitely does not) to 0 (variables appearing or not appearing have no effect on one another)
+            to 1 (if one variable appears the other definitely also does).
+            Entries marked <1 or >-1 have a correlation that is close to being exactingly negative or positive, but is still not quite perfectly so.
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply eda/src/autogluon/eda/auto/simple.py:856]: Fixed

File: docs/tutorials/index.rst [hunk 19]
```diff
@@ -150,6 +150,43 @@ Cloud Training and Deployment
...
+Exploratory Data Analysis
+-------------------------
+
+.. container:: cards
+
+   .. card::
+      :title: Automated Dataset Overview
+      :link: eda/eda-auto-dataset-overview.html
+
+      Tool to get a high-level understanding of datasets including basic statistical information and feature information
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply docs/tutorials/index.rst:162]: Fixed

File: docs/tutorials/eda/index.rst [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+
+This section provides an overview of exploratory data analysis tools.
+
+.. container:: cards
+
+   .. card::
+      :title: Automated Dataset Overview
+      :link: eda-auto-dataset-overview.html
+
+      Tool to get a high-level understanding of datasets including basic statistical information and feature information
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply docs/tutorials/eda/index.rst:12]: fixed

File: docs/tutorials/eda/eda-auto-analyze-interaction.md [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+This tool is made for quick interactions visualization between variables in a dataset. User can specify the variables to
+be plotted on the x, y and hue (color) parameters. The tool automatically picks chart type to render based on the
+detected variable types and renders 1/2/3-way interactions.
+
+This feature can be useful in exploring patterns, trends, and outliers and potentially identify good predictors for the
+task.
+
+## Using Interaction Charts for Missing Values Filling
+
+Let's load titanic dataset:
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-analyze-interaction.md:12]: fixed

File: docs/tutorials/eda/eda-auto-analyze-interaction.md [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+
+```{.python .input}
+import pandas as pd
+
+df_train = pd.read_csv('https://autogluon.s3.amazonaws.com/datasets/titanic/train.csv')
+df_test = pd.read_csv('https://autogluon.s3.amazonaws.com/datasets/titanic/test.csv')
+target_col = 'Survived'
+```
+
+Next we will look at missing data in the variable:
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-analyze-interaction.md:22]: fixed

File: docs/tutorials/eda/eda-auto-analyze-interaction.md [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+
+```{.python .input}
+import autogluon.eda.auto as auto
+
+auto.missing_values_analysis(train_data=df_train)
+```
+
+It looks like there are only two null values in the `Embarked` feature. Let's see what are those two null values:
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-analyze-interaction.md:30]: fixed

File: docs/tutorials/eda/eda-auto-analyze-interaction.md (hunk 1 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-analyze-interaction.md:53]: fixed

File: docs/tutorials/eda/eda-auto-analyze-interaction.md (hunk 2 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-analyze-interaction.md:64]: fixed

File: docs/tutorials/eda/eda-auto-covariate-shift.md (hunk 3 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-covariate-shift.md:35]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 5 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:6]: fixed

File: docs/tutorials/eda/eda-auto-dataset-overview.md (hunk 4 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-dataset-overview.md:32]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 6 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:7]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 7 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:9]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 8 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:10]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 10 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:18]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 9 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:14]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 11 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:22]: fixed

File: docs/tutorials/eda/eda-auto-quick-fit.md (hunk 12 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-quick-fit.md:41]: changed to `we'll keep only a few columns`

File: docs/tutorials/eda/eda-auto-target-analysis.md (hunk 13 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-target-analysis.md:9]: fixed

File: docs/tutorials/eda/eda-auto-target-analysis.md (hunk 14 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-target-analysis.md:10]: replaced with and

File: docs/tutorials/eda/eda-auto-target-analysis.md (hunk 15 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-target-analysis.md:15]: fixed

File: docs/tutorials/eda/eda-auto-target-analysis.md (hunk 16 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-target-analysis.md:39]: This was a left-over from editing. Removed

File: docs/tutorials/eda/eda-auto-target-analysis.md (hunk 17 above)
Comment (gradientsky) [reply docs/tutorials/eda/eda-auto-target-analysis.md:44]: fixed

**Review by gidler — APPROVED** [review]
